crates/executor/src/blockchain_tree/chain.rs

```diff
@@ -0,0 +1,319 @@
+//! Handles substate and list of blocks.
+//! have functions to split, branch and append the chain.
+use crate::{
+    execution_result::ExecutionResult,
+    substate::{SubStateData, SubStateWithProvider},
+};
+use reth_interfaces::{consensus::Consensus, executor::Error as ExecError, Error};
+use reth_primitives::{BlockHash, BlockNumber, SealedBlockWithSenders, SealedHeader, U256};
+use reth_provider::{BlockExecutor, ExecutorFactory, StateProvider};
+use std::collections::BTreeMap;
+
+/// Side chain that contain it state and connect to block found in canonical chain.
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+pub struct Chain {
+    /// Chain substate
+    pub substate: SubStateData,
+    /// Changesets for block and transaction.
+    pub changesets: Vec<ExecutionResult>,
+    /// Blocks in this chain
+    pub blocks: BTreeMap<BlockNumber, SealedBlockWithSenders>,
+}
+
+/// Where does the chain connect to.
```

Comment:
What is the difference between this and `BlockIndices`? They seem to serve the same purpose?

Reply:
It is a simple structure so it becomes more descriptive to use `.number` and `.hash` and not `.0` or `.1`